Fix thread context propagation for JAX-RS async client callbacks with ManagedExecutorService

# JAX-RS 2.1 Async InvocationCallback JNDI Lookup Fails Intermittently on Windows 11

## Describe the bug
When using JAX-RS 2.1 Client with a `ManagedExecutorService` and `InvocationCallback`, JNDI lookups (`java:comp/env`) fail intermittently on Windows 11 with error CWNEN1000E. The callback thread does not receive the servlet's component context, causing `NamingException` when attempting to access JNDI resources.

### Full Stack Trace
```
junit.framework.AssertionFailedError: 2026-03-02-09:56:02:223 ERROR: Caught exception attempting to call test method testClientBuilderSubmitInvocationCallbackViaManagedExecutor on servlet web.jaxrstest.JAXRSExecutorTestServlet
java.lang.Exception: Unexpected failure for InvocationCallback lookup attempt. See cause.
    at web.jaxrstest.JAXRSExecutorTestServlet.testClientBuilderSubmitInvocationCallbackViaManagedExecutor(JAXRSExecutorTestServlet.java:131)
    at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
    at componenttest.app.FATServlet.doGet(FATServlet.java:76)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:686)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:791)
    at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1262)
    at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:751)
    at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:448)
    at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1362)
    at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:5106)
    at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.handleRequest(DynamicVirtualHost.java:328)
    at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:1047)
    at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:293)
    at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1563)
    at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:761)
    at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:721)
    at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:571)
    at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:505)
    at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:365)
    at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:332)
    at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:169)
    at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:77)
    at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:513)
    at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:583)
    at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:967)
    at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1056)
    at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:344)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
    at java.base/java.lang.Thread.run(Thread.java:1516)
Caused by: javax.naming.NamingException: CWNEN1000E: A JNDI operation on a java:comp/env name cannot be completed because the current thread is not associated with a Java Enterprise Edition application component. This condition can occur when the JNDI client using the java:comp/env name does not occur on the thread of a server application request. Make sure that a Java EE application does not run JNDI operations on java:comp/env names within static code blocks or in threads created by that application. Such code does not necessarily run on the thread of a server application request and therefore is not supported by JNDI operations on java:comp/env names.
    at com.ibm.ws.injectionengine.osgi.internal.naming.InjectionJavaColonHelper.getInjectionScopeData(InjectionJavaColonHelper.java:161)
    at com.ibm.ws.injectionengine.osgi.internal.naming.InjectionJavaColonHelper.getInjectionBinding(InjectionJavaColonHelper.java:135)
    at com.ibm.ws.injectionengine.osgi.internal.naming.InjectionJavaColonHelper.getObjectInstance(InjectionJavaColonHelper.java:115)
    at com.ibm.ws.jndi.url.contexts.javacolon.internal.JavaURLContext.lookup(JavaURLContext.java:334)
    at com.ibm.ws.jndi.url.contexts.javacolon.internal.JavaURLContext.lookup(JavaURLContext.java:372)
    at org.apache.aries.jndi.DelegateContext.lookup(DelegateContext.java:149)
    at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
    at web.jaxrstest.JAXRSExecutorTestServlet$1.completed(JAXRSExecutorTestServlet.java:103)
```

## Steps to Reproduce

### 1. Create a JAX-RS 2.1 Client with a ManagedExecutorService

```java
@Resource(lookup = "concurrent/scheduledExecutor")
private ManagedScheduledExecutorService scheduledExecutor;

ClientBuilder clientBuilder = ClientBuilder.newBuilder();
Client client = clientBuilder.executorService(scheduledExecutor).build();
```

### 2. Submit an async request with an InvocationCallback that performs JNDI lookup

```java
String uri = "http://localhost:8010/jaxrsapp/testapp/test/info";

Future<String> future = client.target(uri)
    .request("text/plain")
    .buildGet()
    .submit(new InvocationCallback<String>() {
        @Override
        public void completed(String response) {
            try {
                // This JNDI lookup fails intermittently on Windows 11
                Object resource = new InitialContext().lookup("java:comp/env/concurrent/scheduledExecutorRef");
                // Use the resource...
            } catch (NamingException x) {
                // CWNEN1000E error occurs here
                throw new RuntimeException(x);
            }
        }
        
        @Override
        public void failed(Throwable x) {
            // Handle failure
        }
    });

String result = future.get();
```

### 3. Run on Windows 11
The failure occurs intermittently due to thread scheduling differences.

## Expected behavior
The `InvocationCallback` should have access to the servlet's JNDI context (`java:comp/env`) when using a `ManagedExecutorService`, allowing JNDI lookups to succeed consistently across all platforms (Windows, Mac, Linux).

## Diagnostic information

- **OpenLiberty Version:** JAX-RS 2.1 (affects multiple versions)
- **Affected feature(s):** `jaxrsClient-2.1`, `concurrent-1.0`
- **Java Version:** Java 11+ (tested on Windows 11)
- **Platform:** 
  - Windows 11: Fails intermittently
- **Test Case:** `com.ibm.ws.jaxrs.2.1_fat_executor.JAXRSExecutorTestServlet.testClientBuilderSubmitInvocationCallbackViaManagedExecutor`

### server.xml configuration

```xml
<featureManager>
    <feature>jaxrsClient-2.1</feature>
    <feature>concurrent-1.0</feature>
    <feature>servlet-4.0</feature>
</featureManager>

<managedScheduledExecutorService id="scheduledExecutor" 
                                  jndiName="concurrent/scheduledExecutor"/>
```

## Additional context

### Root Cause Analysis

Through detailed trace log analysis comparing **failing Windows runs** vs **passing Mac runs**, the issue was identified as a race condition in thread context propagation:

#### Windows (FAIL) - Context Capture Timeline
1. **11:04:09:611** Thread `000000c2` (servlet thread): 
   - Captures context: `JEEMetadataContextImpl@7936dc81 jaxrsapp#jaxrsapp.war#web.jaxrstest.JAXRSExecutorTestServlet` **GOOD**

2. **11:04:09:692** Thread `000000d3` (worker thread):
   - Captures and **OVERWRITES** with: `JEEMetadataContextImpl@40154101 default` **BAD**

3. **11:04:09:810** Thread `000000d3` (worker thread):
   - Captures and **OVERWRITES** again with: `JEEMetadataContextImpl@62d6c6f3 default` **BAD**

4. **11:04:09:849** Thread `000000c5` (callback thread):
   - Receives: `JEEMetadataContextImpl@1d557e8c default`  **WRONG CONTEXT**
   - JNDI lookup fails with CWNEN1000E

#### Mac (PASS) - Context Capture Timeline
1. **17:04:17:935** Thread `0000004b` (servlet thread):
   - Captures context: `JEEMetadataContextImpl@5b5779d1 jaxrsapp#jaxrsapp.war#web.jaxrstest.JAXRSExecutorTestServlet`  **GOOD**

2. **17:04:17:937** Thread `0000004b` (same servlet thread):
   - Captures context: `JEEMetadataContextImpl@cadab9d2 jaxrsapp#jaxrsapp.war#web.jaxrstest.JAXRSExecutorTestServlet` **GOOD**

3. **17:04:17:976** Thread `00000047` (callback thread):
   - Receives: `JEEMetadataContextImpl@17a11da3 jaxrsapp#jaxrsapp.war#web.jaxrstest.JAXRSExecutorTestServlet` **CORRECT CONTEXT**
   - JNDI lookup succeeds

### Why Windows Fails

The Windows thread scheduler causes `ManagedExecutorRunnableWrapper.prepare()` to be called multiple times on **different threads**. Later calls on worker threads overwrite the good servlet context captured initially, resulting in the callback receiving a "default" context without JNDI bindings.

### Proposed Fix

Two complementary changes are needed to resolve this issue:

#### 1. Early Context Capture
**File:** `com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/http/HTTPConduit.java`

Add context capture at the beginning of `prepare()` method (line 552):

```java
@Override
public void prepare(Message message) throws IOException {
    // Liberty Change: Capture thread context early for async callbacks with ManagedExecutorService
    // This must be done here on the servlet thread before async operations begin
    AsyncClientRunnableWrapperManager.prepare(message);
    
    // This call can possibly change the conduit endpoint address and
    // protocol from the default set in EndpointInfo that is associated
    // with the Conduit.
    // ... rest of method
}
```

#### 2. Prevent Context Overwriting
**File:** `io.openliberty.cxf.concurrent/src/com/ibm/ws/cxf/concurrent/component/ManagedExecutorRunnableWrapper.java`

Add defensive check in `prepare()` method (line 31):

```java
@Override
public void prepare(Message message) {
    Executor executor = message.getExchange().get(Executor.class);
    if (executor instanceof WSManagedExecutorService) {
        // Only capture context if not already captured to preserve servlet thread context
        // This prevents overwriting good context captured earlier in HTTPConduit.prepare()
        ThreadContextDescriptor existingTcd = message.get(ThreadContextDescriptor.class);
        if (existingTcd == null) {
            final ThreadContextDescriptor tcd = ((WSManagedExecutorService) executor).captureThreadContext(null);
            message.put(ThreadContextDescriptor.class, tcd);
        }
    }
}
```

### How the Fix Works

1. **Early Capture:** When `HTTPConduit.prepare()` is called on the servlet thread, it immediately captures the servlet's thread context including JNDI bindings
2. **Preservation:** When `handleResponseAsync()` later calls `prepare()` again (potentially on a worker thread), the defensive check prevents overwriting the good context
3. **Propagation:** The captured servlet context is wrapped into a `ContextualRunnable` and applied when the callback executes

### Expected Result After Fix

- Context captured **once** on servlet thread with correct JNDI bindings
- Subsequent captures on worker threads are **skipped**
- Callback receives **servlet context** and JNDI lookups succeed
- Test passes **consistently** on Windows 11

## Impact

- **Severity:** Medium-High (intermittent failures, platform-specific)
- **Scope:** Affects JAX-RS 2.1 async clients using `ManagedExecutorService` on Windows
- **User Impact:** Applications using JNDI in JAX-RS callbacks may fail unpredictably on Windows platforms

## Related Files

- `com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/http/HTTPConduit.java`
- `io.openliberty.cxf.concurrent/src/com/ibm/ws/cxf/concurrent/component/ManagedExecutorRunnableWrapper.java`
- `com.ibm.ws.jaxrs.2.1_fat_executor/test-applications/jaxrsapp/src/web/jaxrstest/JAXRSExecutorTestServlet.java`

Stack Trace: 

```
junit.framework.AssertionFailedError: 2025-07-18-12:15:37:082 ERROR: Caught exception attempting to call test method testClientBuilderSubmitInvocationCallbackViaManagedExecutor on servlet web.jaxrstest.JAXRSExecutorTestServlet
java.lang.Exception: Unexpected failure for InvocationCallback lookup attempt. See cause.
	at web.jaxrstest.JAXRSExecutorTestServlet.testClientBuilderSubmitInvocationCallbackViaManagedExecutor(JAXRSExecutorTestServlet.java:131)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at componenttest.app.FATServlet.doGet(FATServlet.java:76)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:686)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:791)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1266)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:754)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:451)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1362)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1078)
	at com.ibm.ws.webcontainer.servlet.CacheServletWrapper.handleRequest(CacheServletWrapper.java:77)
	at com.ibm.ws.webcontainer40.servlet.CacheServletWrapper40.handleRequest(CacheServletWrapper40.java:87)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:978)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:293)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1284)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:500)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:459)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:569)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:503)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:363)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:330)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:169)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:77)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:516)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:586)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:970)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1059)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:301)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:842)
Caused by: javax.naming.NamingException: CWNEN1000E: A JNDI operation on a java:comp/env name cannot be completed because the current thread is not associated with a Java Enterprise Edition application component. This condition can occur when the JNDI client using the java:comp/env name does not occur on the thread of a server application request. Make sure that a Java EE application does not run JNDI operations on java:comp/env names within static code blocks or in threads created by that application. Such code does not necessarily run on the thread of a server application request and therefore is not supported by JNDI operations on java:comp/env names.
	at com.ibm.ws.injectionengine.osgi.internal.naming.InjectionJavaColonHelper.getInjectionScopeData(InjectionJavaColonHelper.java:161)
	at com.ibm.ws.injectionengine.osgi.internal.naming.InjectionJavaColonHelper.getInjectionBinding(InjectionJavaColonHelper.java:135)
	at com.ibm.ws.injectionengine.osgi.internal.naming.InjectionJavaColonHelper.getObjectInstance(InjectionJavaColonHelper.java:115)
	at com.ibm.ws.jndi.url.contexts.javacolon.internal.JavaURLContext.lookup(JavaURLContext.java:334)
	at com.ibm.ws.jndi.url.contexts.javacolon.internal.JavaURLContext.lookup(JavaURLContext.java:372)
	at org.apache.aries.jndi.DelegateContext.lookup(DelegateContext.java:149)
	at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
	at web.jaxrstest.JAXRSExecutorTestServlet$1.completed(JAXRSExecutorTestServlet.java:103)
	at web.jaxrstest.JAXRSExecutorTestServlet$1.completed(JAXRSExecutorTestServlet.java:94)
	at org.apache.cxf.jaxrs.client.JaxrsClientCallback.handleResponse(JaxrsClientCallback.java:87)
	at org.apache.cxf.jaxrs.client.WebClient$ClientAsyncResponseInterceptor.doHandleAsyncResponse(WebClient.java:1038)
	at org.apache.cxf.jaxrs.client.AbstractClient$AbstractClientAsyncResponseInterceptor.handleAsyncResponse(AbstractClient.java:1311)
	at org.apache.cxf.jaxrs.client.AbstractClient$AbstractClientAsyncResponseInterceptor.handleMessage(AbstractClient.java:1281)
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:325)
	at org.apache.cxf.jaxrs.client.ClientMessageObserver.onMessage(ClientMessageObserver.java:56)
	at org.apache.cxf.transport.http.HTTPConduit$WrappedOutputStream.handleResponseInternal(HTTPConduit.java:1945)
	at org.apache.cxf.transport.http.HTTPConduit$WrappedOutputStream$1.run(HTTPConduit.java:1316)
	at com.ibm.ws.threading.internal.PolicyTaskFutureImpl.run(PolicyTaskFutureImpl.java:762)
	at com.ibm.ws.threading.internal.PolicyExecutorImpl.runTask(PolicyExecutorImpl.java:1341)
	at com.ibm.ws.threading.internal.PolicyExecutorImpl$AsyncTask.run(PolicyExecutorImpl.java:220)

```

--- a/dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/transport/http/HTTPConduit.java
+++ b/dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/transport/http/HTTPConduit.java
@@ -1225,6 +1225,8 @@ public abstract class HTTPConduit
 
         @FFDCIgnore(RejectedExecutionException.class)
         protected void handleResponseOnWorkqueue(boolean allowCurrentThread, boolean forceWQ) throws IOException {
+            //Capture thread context before creating the runnable
+            AsyncClientRunnableWrapperManager.prepare(outMessage);
             Runnable runnable = new Runnable() {
                 @Override
                 @FFDCIgnore(Throwable.class)

--- a/dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/transport/http/HTTPConduit.java
+++ b/dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/transport/http/HTTPConduit.java
@@ -92,6 +92,7 @@ import org.apache.cxf.ws.addressing.EndpointReferenceType;
 
 import com.ibm.websphere.ras.Tr;
 import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.cxf.jaxrs20.client.component.AsyncClientRunnableWrapperManager;
 import com.ibm.ws.ffdc.annotation.FFDCIgnore;
 
 /*
@@ -1229,7 +1230,7 @@ public abstract class HTTPConduit
             // Capture thread context before creating the runnable
             AsyncClientRunnableWrapperManager.prepare(outMessage);
              // Liberty Change End
-            Runnable runnable = new Runnable() {
+            Runnable runnable = AsyncClientRunnableWrapperManager.wrap(outMessage, new Runnable() {
                 @Override
                 @FFDCIgnore(Throwable.class)
                 public void run() {
@@ -1246,7 +1247,7 @@ public abstract class HTTPConduit
                         mo.onMessage(outMessage);
                     }
                 }
-            };
+            });
             HTTPClientPolicy policy = getClient(outMessage);
             boolean exceptionSet = outMessage.getContent(Exception.class) != null;
             if (!exceptionSet) {

--- a/dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/transport/http/HTTPConduit.java
+++ b/dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/transport/http/HTTPConduit.java
@@ -1225,8 +1225,10 @@ public abstract class HTTPConduit
 
         @FFDCIgnore(RejectedExecutionException.class)
         protected void handleResponseOnWorkqueue(boolean allowCurrentThread, boolean forceWQ) throws IOException {
-            //Capture thread context before creating the runnable
+            // Liberty change Begin
+            // Capture thread context before creating the runnable
             AsyncClientRunnableWrapperManager.prepare(outMessage);
+             // Liberty Change End
             Runnable runnable = new Runnable() {
                 @Override
                 @FFDCIgnore(Throwable.class)

--- a/dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/http/HTTPConduit.java
+++ b/dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/http/HTTPConduit.java
@@ -1308,6 +1308,10 @@ public abstract class HTTPConduit
 
         @FFDCIgnore(RejectedExecutionException.class) // Liberty Change Start
         protected void handleResponseOnWorkqueue(boolean allowCurrentThread, boolean forceWQ) throws IOException {
+            // Liberty change Begin 
+            // Capture thread context before creating the runnable
+            AsyncClientRunnableWrapperManager.prepare(outMessage);
+            // Liberty Change End
             Runnable runnable = AsyncClientRunnableWrapperManager.wrap(outMessage, new Runnable() {
                 @Override
                 @FFDCIgnore(Throwable.class)

